File: api/app/models/form_template.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class FormTemplate(Base):
    __tablename__ = "form_templates"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name: Mapped[str] = mapped_column(String(255), unique=True)
    description: Mapped[str | None] = mapped_column(String(500), nullable=True)
    fields: Mapped[list] = mapped_column(JSONB, default=list)
    nature_of_work_category: Mapped[str] = mapped_column(String(32), default="general")
    language: Mapped[str] = mapped_column(String(8), default="both")
    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.utcnow, onupdate=datetime.utcnow
    )

File: web/lib/form-template.ts
import { apiJson } from "@/lib/api";
import type { FormFieldSchema } from "@/lib/formSchema";

export type FormTemplateDetail = {
  id: string;
  name: string;
  description: string | null;
  fields: FormFieldSchema[];
  nature_of_work_category: string;
  language: string;
  is_default: boolean;
  created_at: string;
  updated_at: string;
};

export async function fetchFormTemplate(
  templateId: string,
  locale: string,
): Promise<FormTemplateDetail> {
  return apiJson<FormTemplateDetail>(`/form-templates/${templateId}`, {}, locale);
}
